packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -655,14 +655,30 @@ rocksdb::Status ArbCore::saveAssertion(ReadWriteTransaction& tx,
     return rocksdb::Status::OK();
 }
 
-rocksdb::Status ArbCore::reorgToLastMessage(ValueCache& cache) {
-    std::cerr << "Reloading chain to the last message saved"
+rocksdb::Status ArbCore::reorgToLastCheckpoint(ValueCache& cache) {
+    std::cerr << "Reloading chain to the last checkpoint saved"
               << "\n";
 
     return reorgCheckpoints([&](const MachineOutput&) { return true; }, true,
                             cache);
 }
 
+rocksdb::Status ArbCore::reorgToPenultimateCheckpoint(ValueCache& cache) {
+    std::cerr << "Reloading chain to the penultimate checkpoint saved"
+              << "\n";
+
+    return reorgCheckpoints([&](const MachineOutput&) { return true; }, true,
+                            cache);
+    return reorgCheckpoints(
+        [&](const MachineOutput& output) {
+            static int count = 0;
+            count++;
+            // Skip first entry
+            return count > 1;
```

Comment:
I think `count` will be kept across multiple invocations of `reorgToPenultimateCheckpoint`, though I'm not entirely sure. A simpler solution might be to declare `int count = 0;` outside the lambda.

Reply:
:+1: